gitlab/gitlab_exporter.py

```diff
@@ -0,0 +1,127 @@
+    """
+
+    credentials = {'clientId': PORT_CLIENT_ID, 'clientSecret': PORT_CLIENT_SECRET}
+
+    token_response = requests.post(f"{PORT_API_URL}/auth/access_token", json=credentials)
+
+    return token_response.json()['accessToken']
+
+def create_entity(blueprint: str, body: json, access_token: str):
+    """
+    Create new entity for blueprint in Port
+    """
+    
+    headers = {
+        'Authorization': f'Bearer {access_token}',
+        'User-Agent': 'gitlab-exporter',
+    }
+
+    response = requests.post(f"{PORT_API_URL}/blueprints/{blueprint}/entities?upsert=true&merge=true",json=body, headers=headers)
+
+    return response
+
+def get_all_merge_requests_from_gitlab(token: str):
+    merge_requests_url = f"{GITLAB_URL}/groups/{GROUP_ID}/merge_requests"
+
+    response = requests.get(
+        merge_requests_url,
+        headers={"PRIVATE-TOKEN": GITLAB_API_TOKEN},
+    )  
+
+    if response.status_code == 200:
+        merge_requests = response.json()
+
+        for merge_request in merge_requests:
+            entity = {
+                'identifier': str(merge_request['id']),
+                'title': merge_request['title'],
+                'properties': {
+                    'creator': merge_request['author']['username'],
+                    'status': merge_request['state'],
+                    'createdAt': merge_request['created_at'],
+                    'updatedAt': merge_request['updated_at'],
+                    'description': merge_request['description'],
+                    'link': merge_request['web_url']
+                },
+                'relations': {
+                    'microservice': str(merge_request['source_project_id']),
+                }
+            }
+
+            create_entity('mergeRequest', entity, access_token=token)
+    else:
+        print(f"Failed to get merge requests. Status code: {response.status_code}")
+
+
+def get_all_entities_from_gitlab():
+    # Gets all projects from GitLab and create a microservice for each one
+    # Foreach project gets all merge requests from GitLab and create a merge request for each one
+
+    # membership=true will return only the projects that the user is a member of (and not all public projects)
+    api_url = f"{GITLAB_URL}/groups/{GROUP_ID}/projects"
+
+    response = requests.get(
+        api_url,
+        headers={"PRIVATE-TOKEN": GITLAB_API_TOKEN},
+    )  
+
+    if response.status_code == 200:
+        projects = response.json()
+
+        token = get_port_api_token()
+        for project in projects:
+            entity = {
+                'identifier': str(project['id']),
+                'title': project['name'],
+                'properties': {
+                    'description': project['description'],
+                    'url': project['web_url']
+                }
+            }
+
+            create_entity('microservice', entity, access_token=token)
+            get_all_merge_requests_from_gitlab(token)
```

Comment:
The merge requests URL seems to be related to the GROUP_ID and not to a specific project, so I think this function call should not be part of the `for` loop (it will be called again and again for no reason, better to just call it once after the `for` loop has finished and all project entities have been created)